### PR TITLE
feat: Edge Processor

### PR DESCRIPTION
**Pull request by ikheifets-splunk**

### CODE REVIEW

File: package/etc/conf.d/conflib/_common/t_templates.conf [hunk 1]
```diff
@@ -138,7 +138,6 @@ template t_JSON_5424_SDATA {
 
 template t_splunk_hec {
     template('$(format-json
-        time=$(if ("${.netsource.sc4s_use_recv_time}" eq "yes") "$R_UNIXTIME" "$S_UNIXTIME")
```
Comment (ikheifets-splunk) [inline package/etc/conf.d/conflib/_common/t_templates.conf:141]: When you sending to EP json with `time` it's ignoring it

File: package/sbin/entrypoint.sh [hunk 2]
```diff
@@ -175,13 +175,13 @@ then
   SC4S_DEST_SPLUNK_HEC_EVENTS_INDEX=$(cat $SC4S_ETC/conf.d/local/context/splunk_metadata.csv | grep ',index,' | grep sc4s_events | cut -d, -f 3)
   export SC4S_DEST_SPLUNK_HEC_EVENTS_INDEX=${SC4S_DEST_SPLUNK_HEC_EVENTS_INDEX:=main}
 
-  if curl -s -S ${NO_VERIFY} "${HEC}?/index=${SC4S_DEST_SPLUNK_HEC_FALLBACK_INDEX}" -H "Authorization: Splunk ${SC4S_DEST_SPLUNK_HEC_DEFAULT_TOKEN}" -d '{"event": "HEC TEST EVENT", "sourcetype": "sc4s:probe"}' 2>&1 | grep -v '{"text":"Success"'
+  if curl -s -S ${NO_VERIFY} "${HEC}?/index=${SC4S_DEST_SPLUNK_HEC_FALLBACK_INDEX}" -H "Authorization: Splunk ${SC4S_DEST_SPLUNK_HEC_DEFAULT_TOKEN}" -d '{"event": "HEC TEST EVENT", "sourcetype": "sc4s:probe"}' 2>&1 | grep -v '{"text": "Success"'
```
Comment (ikheifets-splunk) [inline package/sbin/entrypoint.sh:178]: EP returning 

`{"text": "Success"}`

insead of:

{"text":"Success"}


the difference in whitespaces

File: package/etc/conf.d/sources/source_syslog/plugin.jinja [hunk 3]
```diff
@@ -104,6 +104,7 @@ source s_{{ port_id }} {
             {%- endfor %}
         };
 
+        {% if not ignore_parsing %}
```
Comment (ikheifets-splunk) [inline package/etc/conf.d/sources/source_syslog/plugin.jinja:107]: adding new mode for stream raw message without assigning `sourcetype` and `index`

File: docs/edge_processor.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Edge Processor integration guide (Experimental)
```
[the remark this answers is not in the document]
Comment (rjha-splunk) [reply docs/edge_processor.md:1]: It should be mentioned in the PR , this is highly experimental feature, only use based on suggestion/recommendation by Splunk

File: docs/edge_processor.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+ 1. Use IP of EP instance as HEC URL
+ 2. Use token from EP Global Settings
+ 3. Use EP API format: t_edge_hec
+
+### Docker:
+
+Update `env_file`:
+```
+SC4S_DEST_SPLUNK_HEC_DEFAULT_URL=http://x.x.x.x:8088
```
[the remark this answers is not in the document]
Comment (rjha-splunk) [reply docs/edge_processor.md:46]: Lets keep it same as standard.

[the remark this answers is not in the document]
Comment (rjha-splunk) [reply docs/edge_processor.md:46]: what if customer need to send to Splunk and Send to EP as well, so we need to introduce a variable specific to EP

File: package/etc/conf.d/conflib/_common/t_templates.conf [hunk 6]
```diff
@@ -138,7 +138,7 @@ template t_JSON_5424_SDATA {
...
+        time=$(if ("${.netsource.sc4s_use_recv_time}" eq "yes") ${R_UNIXTIME} ${S_UNIXTIME})
```
Comment (rjha-splunk) [inline package/etc/conf.d/conflib/_common/t_templates.conf:141]: Can we introduce a env variable , which can be enabled to change this to INT in place of STR and introduce a loop , Please check the feasibility..

File: docs/edge_processor.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    Exmaple cert
+    -----END CERTIFICATE-----
+  ca: |
+    -----BEGIN CERTIFICATE-----
+    Example ca
+    -----END CERTIFICATE-----
+```
+
+  3. Encrypt your `secrets.yaml` using `ansible-vault`.
+  4. Add IP of cluster nodes to inventoey file `ansible/inventory/inventory_microk8s_ha.yaml`
```
Comment (rjha-splunk) [inline docs/edge_processor.md:109]: inventory

**Review by mstopa-splunk — APPROVED** [review]

File: package/etc/conf.d/conflib/_common/t_templates.conf (hunk 6 above)
Comment (rjha-splunk) [reply package/etc/conf.d/conflib/_common/t_templates.conf:141]: kindly mention it in PR merge doc

**Review by rjha-splunk — COMMENTED** [review]
Kindly change the PR merge doc as commented.

**Pull request merged**